blocks/library/embed/index.js

```diff
@@ -216,14 +229,20 @@ function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, k
 		},
 
 		save( { attributes } ) {
-			const { url, caption, align } = attributes;
+			const { url, caption, align, type, providerNameSlug } = attributes;
 
 			if ( ! url ) {
 				return;
 			}
 
+			const embedClassName = classnames( 'wp-block-embed', {
```

Comment:
This class should be auto-generated.

Reply:
Technically yes, but this is only true on the core/embed block. If the user chooses one of the flavored embed blocks, e.g. core/embed-youtube, the class will change to wp-block-embed-youtube and the `wp-block-embed` class will not be available anymore. 

You could mitigate that in CSS with unqualified attribute selectors (as described in #4107), but that would be at a huge cost for frontend rendering performance. 

So I am currently thinking about how to avoid class duplication on core/embed, the behavior on the flavored embeds is as intended.

Reply:
Since for a YouTube embed, the `wp-block-embed-youtube` class is applied, isn't the `is-provider-youtube` class then redundant?

Reply:
Non-blocking: We'd probably expect the fixture markup to be different for most embeds, i.e. including `providerNameSlug` and `type` attributes / classes:

```
<!-- wp:core-embed/youtube {"url":"https://www.youtube.com/watch?v=c-WDHG6rrdU","align":"right","type":"video","providerNameSlug":"youtube"} -->
<figure class="wp-block-embed-youtube wp-block-embed is-alignright is-type-video is-provider-youtube">
    https://www.youtube.com/watch?v=c-WDHG6rrdU
</figure>
<!-- /wp:core-embed/youtube -->
```

Reply:
The 'is-provider-' classes are needed for two reasons: 

1) We cannot be sure, that the user currently uses the correct block. They can use the wp-block-embed to yield the same result as with wp-block-embed-youtube. That was the whole premise of this PR.

2) The provider name can be different for some providers like the wordpress embed, where it describes the URL where the content is coming from. That adds another level of description for theme editors. (Like the ability to style the WP Embed block differently by provider.)

As for the fixture markup, how would I approach that? Do I have to add that by hand?